drivers/spi/spi_max32.c

```diff
@@ -242,12 +272,52 @@ static int spi_max32_transceive(const struct device *dev)
 	dfs_shift = spi_max32_get_dfs_shift(ctx);
 
 	len = spi_context_max_continuous_chunk(ctx);
+
+#ifdef CONFIG_SPI_RTIO
+	switch (sqe->op) {
+	case RTIO_OP_RX:
+		len = sqe->rx.buf_len;
+		data->req.rxData = sqe->rx.buf;
+		data->req.rxLen = sqe->rx.buf_len;
+		data->req.txData = NULL;
+		data->req.txLen = len >> dfs_shift;
+		break;
+	case RTIO_OP_TX:
+		len = sqe->tx.buf_len;
+		data->req.rxData = NULL;
+		data->req.rxLen = 0;
+		data->req.rxData = data->dummy;
+		data->req.txLen = len >> dfs_shift;
+		break;
+	case RTIO_OP_TINY_TX:
+		len = sqe->tiny_tx.buf_len;
+		data->req.txData = (uint8_t *)sqe->tiny_tx.buf;
+		data->req.rxData = data->dummy;
+		data->req.txLen = len >> dfs_shift;
+		data->req.rxLen = len >> dfs_shift;
+		break;
+	case RTIO_OP_TXRX:
+		len = sqe->txrx.buf_len;
+		data->req.txData = (uint8_t *)sqe->txrx.tx_buf;
+		data->req.rxData = sqe->txrx.rx_buf;
+		data->req.txLen = len >> dfs_shift;
+		data->req.rxLen = len >> dfs_shift;
+		if (!data->req.rxData) {
```

Comment:
This shouldn't be needed actually

Reply:
is this something you plan on addressing in this PR still? Would be good to fix this small nit.